upgrade/scripts/upgrade/rollout-restart.sh

```diff
@@ -0,0 +1,259 @@
+
+  echo "Checking the status of new pods..."
+  echo "$new_pods"
+
+  # Check if all pods are in Running state and have the correct proxyv2 version
+  all_correct=true
+  for pod in $(echo "$new_pods" | awk '{print $1}' | grep -v NAME); do
+    status=$(kubectl get pod "$pod" -n "$NAMESPACE" -o jsonpath='{.status.phase}')
+    image=$(kubectl get pod "$pod" -n "$NAMESPACE" -o jsonpath='{.spec.containers[*].image}')
+
+    if [[ $status != "Running" ]]; then
+      echo "Pod $pod is not in Running state. Current state: $status"
+      all_correct=false
+    fi
+
+    if [[ $image != *"$CORRECT_VERSION"* ]]; then
+      echo "Pod $pod does not have the correct proxyv2 version. Current version: $image"
+      all_correct=false
+    fi
+  done
+
+  if [ "$all_correct" = true ]; then
+    echo "All pods are running with the correct proxyv2 version."
+  else
+    echo "Some pods are not running or do not have the correct proxyv2 version."
+  fi
+else
+  echo "No pods were deleted. Skipping wait."
+fi
+
+# The fuctionality of the script is:
+# This script is to restart the vault pods separately.
+# Because vault was not upgraded in this release that is why we need this script.
+# Get all istio-injection=enabled namespaces.
+# For each namespace we are keeping the record of the sts/deploys/ds which we need to restart by following way:
+#   Get all the pods in the namespace, and check istio image version.
+#   if image is not in 1.19.10 then
+#       Keep the name of the corresponding ds/sts/deploy in a list.
+#   Else move to another pod.
+# After iterating over all the namespaces we will have the list of sts/ds/deploys that needs to be restarted.
+# We will restart the resources from the list that we have made in above steps.
+# We will wait for 2-3 mins and check the rollout status for the resources.
+
+# Function to check if any container in a pod has the latest Istio image version
+check_pod_istio_versions() {
+  local namespace=$1
+  local pod=$2
+  local images
+  images=$(kubectl get pod "$pod" -n "$namespace" -o jsonpath="{.spec.containers[*].image}")
+
+  # Check if any of the images is the latest Istio version
+  if echo "$images" | grep -q "istio/pilot:1.19.10" || echo "$images" | grep -q "istio/proxyv2:1.19.10"; then
+    return 0 # Pod has the latest Istio versions
+  else
+    return 1 # Pod does not have the latest Istio versions
+  fi
+}
+
+# Function to determine the controlling resource of a pod
+get_controlling_resource() {
+  local namespace=$1
+  local pod=$2
+  owner_references=$(kubectl get pod "$pod" -n "$namespace" -o jsonpath="{.metadata.ownerReferences[0].kind}/{.metadata.ownerReferences[0].name}")
+
+  if [[ -n $owner_references ]]; then
+    echo "$owner_references"
+  else
+    # Fallback to describe to get owner information if needed
+    controlling_resource=$(kubectl describe pod "$pod" -n "$namespace" | grep -E "Controlled By" | awk -F: '{print $2}' | xargs)
+    echo "$controlling_resource"
+  fi
```

Comment:
I don't see why there is a need to have two methods, they are both pulling the exact same information, all this could be just a one line function:
```
get_controlling_resource() {
    kubectl get pod "$2" -n "$1" -o jsonpath="{.metadata.ownerReferences[0].kind}/{.metadata.ownerReferences[0].name}"
}
```

To test this I ran this script on mug to compare the two methods:
```
kubectl get pods -A -o jsonpath='{range .items[*]}{.metadata.namespace} {.metadata.name}{"\n"}{end}' | \
while read namespace pod
do
        by_get=$(kubectl get pod "$pod" -n "$namespace" -o jsonpath="{.metadata.ownerReferences[0].kind}/{.metadata.ownerReferences[0].name}")
        by_describe=$(kubectl describe pod "$pod" -n "$namespace" | grep -E "Controlled By" | awk -F: '{print $2}' | xargs)
        if [ "${by_get}" != "${by_describe}" ]; then
            echo "mismatch: ${namespace} ${pod}"
            echo "  ${by_get}"
            echo "  ${by_describe}"
        fi
done
```
and it did not find a single instance where the `describe` returned results when the `get` did not.